ui/client/containers/NussknackerInitializer.tsx

```diff
@@ -136,6 +140,22 @@ class NussknackerInitializer extends React.Component<Props, State> {
 
       const queryHashParams = queryString.parse(this.props.history.location.hash)
       if (settings.implicitGrantEnabled === true && queryHashParams.access_token) {
+        if (settings.jwtAuthServerPublicKey) {
+          let verificationPassed = jwt.verify(queryHashParams.access_token, settings.jwtAuthServerPublicKey)
+          if (verificationPassed) {
+            if (queryHashParams.id_token)
+              verificationPassed = jwt.verify(queryHashParams.id_token, settings.jwtAuthServerPublicKey, {nonce: window.localStorage.getItem("nonce")})
+            else if (settings.jwtIdTokenNonceVerificationRequired === true) {
+              console.log("Warning: jwt.idTokenNonceVerificationRequired=true but id_token missing in the auth server response")
+              verificationPassed = undefined
+            }
+          }
+          if (!verificationPassed) {
+            console.log("JWT verification failed")
```

Comment:
Do we need this console.log?

Reply:
It is not necessary nor helpful (information carried by this log is useless).
But the warning `console.warn("Warning: jwt.idTokenNonceVerificationRequired=true but id_token missing in the auth server response")` might be insufficient for a user that configures nk oauth2 settings.
There could be a helpful warning for oauth2 configuration purpose; I mean ensuring that a received access token matches JWT format (without signature verification). Access token format depends on auth server configuration and request parameters.

Reply:
So, mayby let's do it by console.warn?:)